### PR TITLE
Memcached instrumentation

### PR DESCRIPTION
**Pull request by willianpc**
Nothing extraordinary here, however a few "new" ideas were added that may require attention during the code review:

 * The test app is using `node-fetch` instead of `request-promise` for the HTTP client calls
 * In the integration test, the list of operations to verify were split into sequential and non sequential (individual promises and Promise.all calls, basically). So the code looks bigger, but the test runs faster
 * A few typings were added, as well as some minor refactoring: converting `arguments` to a real array moved to it's own reusable function

### CODE REVIEW

**Review by basti1302 — APPROVED** [review]
Looks good to me.

A few nitpicks here and there, as always :-)

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 - Reporting correct SQS entry span duration
 - Reporting SQS-Consumer processing error messages to the agent
+- Instrumentation of memcached
```
[the remark this answers is not in the document]
Comment (willianpc) [reply CHANGELOG.md:6]: Was missing one rebase

[the remark this answers is not in the document]
Comment (basti1302) [reply CHANGELOG.md:6]: Ah, nevermind then. I should have checked main 🤦

File: packages/collector/test/tracing/database/memcached/test.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                    .sendRequest({
+                      method: 'GET',
+                      path: `${apiPath}`,
+                      simple: withError === false
+                    })
+                    .then(response => {
+                      verify(appControls, response, apiPath, withError, op).then(resolve).catch(reject);
+                    })
+                    .catch(reject);
+                });
```
[the remark this answers is not in the document]
Comment (willianpc) [reply packages/collector/test/tracing/database/memcached/test.js:110]: Good point. Will refactor

File: packages/collector/test/tracing/database/memcached/test.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+  describe('tracing enabled, no suppression', function () {
+    const appControls = new ProcessControls({
+      appPath: path.join(__dirname, 'app'),
+      port: 3215,
+      useGlobalAgent: true,
+      env: {}
+    });
+
+    ProcessControls.setUpHooksWithRetryTime(retryTime, appControls);
```
[the remark this answers is not in the document]
Comment (willianpc) [reply packages/collector/test/tracing/database/memcached/test.js:72]: First argument is always a number, so I will keep this unchanged, unless you suggest some fixed number

[the remark this answers is not in the document]
Comment (basti1302) [reply packages/collector/test/tracing/database/memcached/test.js:72]: That's why I said `ProcessControls.setUpHooks(appControls)` would suffice. Different method than `ProcessControls.setUpHooksWithRetryTime` (which indeed requires you to pass a retry time).

[the remark this answers is not in the document]
Comment (willianpc) [reply packages/collector/test/tracing/database/memcached/test.js:72]: Sorry, I missed that the method was different

**Pull request merged**